### PR TITLE
Iris_cluster module and GUI

### LINKED ISSUE #258: Clustering Module only accepts .json file but not JSON object .
While developing a GUI for ( Iris Search + Cluster), I was using Maryam's webAPI module and found out that we need to save the result of Iris into a .JSON file and then Call Cluster API supplying JSON file.

**Problems** 
- This isn't a scalable solution when deploying web app (because we have to create JSON files for each user and for each query). 

**Proposed Solution**
- Have a unified api which can supply the result of Iris directly to Cluster as JSON object to process data at server side 

Please assign this task to me, 
After completing this task I will be developing a React UI which will consume this API and Layout visualization of Clusters

### PR DESCRIPTION
**Pull request by moshaad7**
Fixes #258 

#### Resolves
Resolves the problem of needing to store result of iris_module in a .json file and then supplying it to cluster_module
And adding a gui for iris_cluster web interface

#### Changes

- Added a new module called **iris_cluster** which takes a query and compute iris_search_result followed by finding cluster_result and then return both results
- Added a flask GUI structure for iris_cluster web interface  

#### Checklist

- [x] I have read the Contribution & Best practices Guideline.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] The acceptance, integration, unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by moshaad7** [comment]
@saeeddhqan @k4u5h1k  please review it. 
I have tested it thoroughly, there were some issues with saving results to gather.dat file. I have fixed them by **Resetting options for output as True**

**Comment by saeeddhqan** [comment]
continue.

**Comment by moshaad7** [comment]
Hii @saeeddhqan thank you for being patient, I have added a flask gui structure. Please review it so that I can go ahead. 

**User guide**
- `python maryam.py -e web api  `
-  visit `http://127.0.0.1:1313/iris_cluster`

**Screenshot**
Screenshot 2021-12-27 at 9 25 14 PM

**Comment by saeeddhqan** [comment]
create your files(js,css,html,..) in core/web/cluster_iris directory.

**Comment by saeeddhqan** [comment]
You only have two days to review the module and test it.

**Comment by moshaad7** [comment]
Hii @saeeddhqan , My Sincere apology for the delay in the response. 
I have checked out your changes and added some more to make iris_cluster_module functional. 
I have tested it for both web and cui, it's working fine. 
Please review it

File: maryam/modules/iris/iris_cluster.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	mode = self._mode
+	# Computing iris search result
+	self._mode = 'api_mode'
+	iris_search_result = self.opt_proc('iris', args=f'-q "{query}" --api', output='web_api')
+	# Computing cluster results
+	clusterer = self.cluster(iris_search_result)
+	output = {'results': clusterer.perform_clustering()}
+
+	self._mode = mode
+	# Resetting options for iris_search_module
```
Comment (moshaad7) [inline maryam/modules/iris/iris_cluster.py:41]: Without resetting options, we are getting wrong command in the web output. I request you to please consider it

**Comment by saeeddhqan** [comment]
Shall I merge?

**Comment by moshaad7** [comment]
Yes @saeeddhqan, It's tested and ready to be merged from my side 
 My only concern is regarding wether to return Iris search results or not ? 
 In my opinion it will be good for performance to return both iris search result and cluster result

**Comment by saeeddhqan** [comment]
please add numpy and sklearn to the requirements

**Pull request merged**